src/main/java/io/neonbee/data/DataContext.java

```diff
@@ -96,6 +98,59 @@ public interface DataContext {
      */
     <T> T remove(String key);
 
+    /**
+     * Arbitrary response data of this context, which is passed backwards from callee to caller.
+     *
+     * @return all the context response data as a map.
+     */
+    Map<String, Object> responseData();
+
+    /**
+     * Merges the response data of a given map into the existing response context data map.
+     * <p>
+     * Note: This method does not perform a deep merge operation, but overrides already existing elements w/o merging
+     *
+     * @param data the response data to merge, must be compatible to {@link JsonObject#JsonObject(Map)}
+     * @return a reference to this DataContext for chaining
+     */
+    DataContext mergeResponseData(Map<String, Object> data);
```

Comment:
Same here? Better fit into DataContextImpl, correct? No user should call this, this is only to be called by the framework